### PR TITLE
Refactor probing to reuse client

### LINKED ISSUE #9136: Reuse client when doing http probing
The client that should be reused is on line 246

https://github.com/cilium/cilium/blob/9a9aa413d4b8fde1c1efd14e373f75026383d437/pkg/health/server/prober.go#L235-L246

### PR DESCRIPTION
**Pull request by borancar**
Rather than recreate the client every single time, just have a global http
client and pass the host to GetHello.

Fixes: #9136

Signed-off-by: Boran Car <boran.car@gmail.com>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/9161)

### CODE REVIEW

**Review by gandro — APPROVED** [review]

**Comment by gandro** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

File: cilium-health/launch/endpoint.go [hunk 1]
```diff
@@ -347,11 +347,11 @@ func LaunchAsEndpoint(baseCtx context.Context, owner regeneration.Owner, n *node
 
 	// Initialize the health client to talk to this instance. This is why
 	// the caller must limit usage of this package to a single goroutine.
-	client, err := probe.NewClient("http://" + net.JoinHostPort(healthIP.String(), fmt.Sprintf("%d", healthDefaults.HTTPPathPort)))
-	if err != nil {
+	client := &Client{host: "http://" + net.JoinHostPort(healthIP.String(), fmt.Sprintf("%d", healthDefaults.HTTPPathPort))}
+	if err = client.PingEndpoint(); err != nil {
```
Comment (borancar) [inline cilium-health/launch/endpoint.go]: This initial ping is probably the cause for failing preflight checks. I didn't see it locally when running the K8s suite but things could just be starting up quicker for me. Note that NewClient only returned errors for incorrect URLs so it never checked the ability to establish a connection before.

**Comment by borancar** [comment]
Misnomer in the commit, should have been "Do not ping during LaunchAsEndpoint". Could you request retesting please?

**Comment by gandro** [comment]
test-me-please

**Review by gandro — APPROVED** [review]

**Pull request merged**